### ISSUE TITLE
CI failing on some JDK 25-ea builds

### ISSUE THREAD

**Issue by thisisalexandercook** [eisop/checker-framework#1219]
The [actions/setup-java](https://github.com/actions/setup-java) action has been selecting the Temurin-Hotspot  jmods archive instead of the full JDK archive since the release of  build JDK 25 ea +21. Since a jmod bundle contains no binaries, subsequent steps in CI fail (example run [here](https://github.com/eisop/checker-framework/pull/1218/checks)).

Logs from an incorrect download:

``` bash
Installed distributions
  Trying to resolve the latest version from remote
  Resolved latest version as 25.0.0+22.0.ea
  Trying to download...
  Downloading Java 25.0.0+22.0.ea (Temurin-Hotspot) from https://github.com/adoptium/temurin25-binaries/releases/download/jdk-25%2B22-ea-beta/OpenJDK-jmods_x64_linux_hotspot_25_22-ea.tar.gz ...
```

Logs from a correct download:

```bash
Installed distributions
  Trying to resolve the latest version from remote
  Resolved latest version as 25.0.0+20.0.ea
  Trying to download...
  Downloading Java 25.0.0+20.0.ea (Temurin-Hotspot) from https://github.com/adoptium/temurin25-binaries/releases/download/jdk-25%2B20-ea-beta/OpenJDK-jdk_x64_linux_hotspot_25_20-ea.tar.gz ...
```

Upstream discussion: actions/setup-java#804

**Comment by Ao-senXiong** [comment]
@thisisalexandercook  Thanks for figuring out the issue, can you create a PR that disables JDK-25 for now? We can revert that PR after the issue from GitHub Action is fixed.

**Comment by wmdietl** [comment]
Thanks for following up on this and glad it's been fixed!
